### PR TITLE
Update support branch version number to 1.14.5-1

### PR DESCRIPTION
**Pull request by glennsong09**

### CODE REVIEW

File: c++/src/cpp_doc_config [hunk 1]
```diff
@@ -38,7 +38,7 @@ PROJECT_NAME           =
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = "1.14.4-1, currently under development"
+PROJECT_NUMBER         = "1.14.5-1, currently under development"
```
Comment (bmribler) [inline c++/src/cpp_doc_config:41]: Is 1.14.5 correct?

Comment (glennsong09) [reply c++/src/cpp_doc_config:41]: According to the release process document, it is correct.

Comment (byrnHDF) [reply c++/src/cpp_doc_config:41]: Now that the release branch is created

**Review by byrnHDF — APPROVED** [review]

File: c++/src/cpp_doc_config (hunk 1 above)
Comment (lrknox) [reply c++/src/cpp_doc_config:41]: 1.14.5 is the correct version for the hdf5_1_14 branch - for the next 1.14 release.  The version in the release branch hdf5_1_14_4 is 1.14.4-2 until the -2 is removed for the actual release.

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**